index.js

```diff
@@ -0,0 +1,183 @@
+"use strict";
+
+const path = require('path');
+const runSequence = require('run-sequence');
+const streamqueue = require('streamqueue');
+const remoteSrc = require('gulp-remote-src');
+const concat = require('gulp-concat');
+const URL = require('url');
+const karmaServer = require('karma').Server;
+
+class Erik {
+  /**
+   * @param {Object} options
+   * @param {Object} options.gulp - The gulp instance with which to register Erik's tasks.
+   * @param {Boolean} [options.watch] - When watch is set, Erik will watch for changes to your
+   * bundled spec and re-run the test suite when changes occur.
+   * @param {String[]} [options.taskDependencies] - Names of tasks to be run before any Erik
+   * processing is done. Useful for registering your spec-building/processing tasks as dependencies
+   * of Erik's testing task. These tasks will be run in serial as passed.
+   * @param {String[]} [options.localDependencies] - Local dependencies to be bundled alongisde your
+   * remote dependencies. Glob strings. Useful for including your bower-installed packages, for
+   * example.
+   * @param {String[]} [options.remoteDependencies] - URLs corresponding to remote dependencies.
+   * @param {String} options.bundledSpecPath - Path to your bundled and processed test suite.
+   * @param {Object} [options.karmaConfig]
+   * @param {Number} [options.karmaConfig.port=9876] - Port on which to run the Karma server.
+   * @param {String} [bundlePath] - Base bath to use for Erik's bundled files. A directory named
+   * `erik` will be created here.
+   */
+  constructor(options) {
+    this._gulp = options.gulp;
+    this._watch = options.watch || false;
+    this._taskDependencies = options.taskDependencies || [];
+    this._localDependencies = options.localDependencies || [];
+    this._remoteDependencies = options.remoteDependencies || [];
+    this._bundledSpecPath = options.bundledSpecPath;
+    this._port = (options.karmaConfig && options.karmaConfig.port) || 9876;
```

Comment:
I see you document this above but let's also specify it in the README. Lots of people won't move beyond the usage example unless they have to